corehq/apps/integration/payments/filters.py

```diff
@@ -0,0 +1,62 @@
+from django.utils.translation import gettext as _
+
+from corehq.apps.es import UserES
+from corehq.apps.reports.filters.base import BaseSingleOptionFilter, BaseMultipleOptionFilter
+from corehq.apps.integration.payments.services import get_payment_batch_numbers_for_domain
+
+
+class PaymentVerificationStatusFilter(BaseSingleOptionFilter):
+    slug = 'payment_verification_status'
+    label = _("Verification Status")
+    default_text = _('Show all')
+
+    verified = 'verified'
+    unverified = 'unverified'
+    options = [
+        (verified, _("Verified")),
+        (unverified, _("Unverified")),
+    ]
+
+
+class BatchNumberFilter(BaseSingleOptionFilter):
+    slug = "batch_number"
+    label = _("Batch number")
+    default_text = _("Show all")
+
+    @property
+    def options(self):
+        batch_numbers = get_payment_batch_numbers_for_domain(self.domain)
+        return [
+            (batch_number, batch_number) for batch_number in batch_numbers
+        ]
+
+
+class PaymentVerifiedByFilter(BaseMultipleOptionFilter):
+    slug = 'verified_by'
+    label = _('Verified by')
+    default_text = _('Show all')
+
+    @property
+    def options(self):
+        query = UserES().domain(self.domain).web_users()
+        return [
+            user_details for user_details in query.values_list('username', 'username')
+        ]
+
+
+class PaymentStatus(BaseSingleOptionFilter):
+    slug = 'payment_status'
+    label = _('Payment status')
+    default_text = _('Show all')
+
+    submitted = 'submitted'
+    not_submitted = 'not_submitted'
```

Comment:
nit: is this same as "pending"?

Reply:
"Submitted" is the same as "pending". 
"Not submitted" is for those cases for which payments have not been requested. 
"Submission failed" is for when the request was made but it could not be completed.

I'm open to update if you think different wording is needed.

Reply:
> Okay, is there anything more we do after the payments are submitted?

Not at the moment. I think we could/should add a callback which will be used to notify HQ of the final payment status (see SC-4376 in the backlog) if we care about that.

The statuses as of today will only show if the payment request has been successful or not, or have not been made.

> so the options would be

So "pending" here refers to any payment that has not been requested, right? Would that be any unverified payment or do we need to update the status to "pending" once it's verified but the payment request has not been made?

Reply:
Yoliswa also raised the same point as you about the pending status, so I'll update.

Reply:
@mkangia See latest commits.